js/ui/windowManager.js

```diff
@@ -310,6 +311,149 @@ var EOSShellWobbly = new Lang.Class({
     }
 });
 
+// ControllableZoomAnimation
+//
+// A "metaclass" for a zoom animation.
+const ControllableZoomAnimation = new Lang.Class({
+    Name: 'ControllableZoomAnimation',
+    Implements: [ AnimationsDbus.ServerEffectBridge ],
+    Extends: GObject.Object,
+    Properties: {
+        length: GObject.ParamSpec.uint('length',
+                                       'Length',
+                                       'How long the animation lasts',
+                                       GObject.ParamFlags.READWRITE |
+                                       GObject.ParamFlags.CONSTRUCT,
+                                       0,
+                                       10000,
+                                       200)
+    },
+
+    vfunc_get_name: function() {
+        return 'zoom';
+    },
+
+    createActorPrivate: function(actor) {
+        let effect = new AttachedZoomAnimation({ bridge: this, enabled: false });
+        actor.add_effect_with_name('endless-animation-zoom', effect);
+        return effect;
+    }
+});
+
+// AttachedZoomAnimation
+//
+// A zoom animation as attached to an actor.
+const AttachedZoomAnimation = new Lang.Class({
+    Name: 'AttachedZoomAnimation',
+    Extends: EndlessShellFX.Affine,
+    Implements: [ AnimationsDbus.ServerSurfaceAttachedEffect ],
+    Properties: {
+        'bridge': GObject.ParamSpec.object('bridge',
+                                           'Zoom Animation Bridge',
+                                           'The bridge to the settings',
+                                           GObject.ParamFlags.READWRITE |
+                                           GObject.ParamFlags.CONSTRUCT_ONLY,
+                                           ControllableZoomAnimation)
+    },
+
+    _init: function(params) {
+        this.parent(params);
+        this._enabledId = -1;
+    },
+
+    _startAnimation: function(animation, done) {
+        this.transform_animation = animation;
+        this.set_enabled(true);
+
+        this._enabledId = this.connect('notify::enabled', () => {
+            if (!this.enabled) {
+                done();
+                this.disconnect(this._enabledId);
+                this.enabledId = -1;
```

Comment:
Assign to 0, as connecting to a signal returns `gulong`. I know this is Js, but still...

Reply:
Indeed. Also fixed all instances of these (also, assigning to the wrong variable, `this.enabledId` vs `this._enabledId`